app/assets/javascripts/helpers/profile_photo_uploader.js

```diff
@@ -8,52 +8,67 @@ Diaspora.ProfilePhotoUploader.prototype = {
+      validation: {
+        allowedExtensions: ["jpg", "jpeg", "png"],
+        sizeLimit: 4194304
       },
+      request: {
+        endpoint: Routes.photos(),
+        params: {
+          /* eslint-disable camelcase */
+          authenticity_token: $("meta[name='csrf-token']").attr("content"),
+          /* eslint-enable camelcase */
+          photo: {"pending": true, "aspect_ids": "all", "set_profile_photo": true}
+        },
+        paramsInBody: false
+      },
+      button: document.getElementById("file-upload"),
 
       messages: {
         typeError: Diaspora.I18n.t("photo_uploader.invalid_ext"),
         sizeError: Diaspora.I18n.t("photo_uploader.size_error"),
         emptyError: Diaspora.I18n.t("photo_uploader.empty")
       },
 
-      onSubmit: function() {
-        $("#file-upload").addClass("loading");
-        $("#profile_photo_upload").find(".avatar").addClass("loading");
-        $("#file-upload-spinner").removeClass("hidden");
-        $("#fileInfo").show();
-      },
+      callbacks: {
+        onProgress: function(id, fileName, loaded, total) {
+          var progress = Math.round(loaded / total * 100);
+          $("#fileInfo").text(fileName + " " + progress + "%");
+        },
+        onSubmit: function() {
+          $("#file-upload").addClass("loading");
+          $("#profile_photo_upload").find(".avatar").addClass("loading");
+          $("#file-upload-spinner").removeClass("hidden");
+          $("#fileInfo").show();
+        },
+        onComplete: function(id, fileName, responseJSON) {
+          $("#file-upload-spinner").addClass("hidden");
+          $("#fileInfo").text(Diaspora.I18n.t("photo_uploader.completed", {"file": fileName}));
+          $("#file-upload").removeClass("loading");
 
-      onComplete: function(_id, fileName, responseJSON) {
-        $("#file-upload-spinner").addClass("hidden");
-        $("#fileInfo").text(Diaspora.I18n.t("photo_uploader.completed", {"file": fileName}));
-        $("#file-upload").removeClass("loading");
-
-        /* flash message prompt */
-        var message = Diaspora.I18n.t("photo_uploader.looking_good");
-        if (app.flashMessages) { app.flashMessages.success(message); }
-
-        var id = responseJSON.data.photo.id;
-        var url = responseJSON.data.photo.unprocessed_image.url;
-        var oldPhoto = $("#photo_id");
-        if (oldPhoto.length === 0) {
-          $("#update_profile_form").prepend("<input type='hidden' value='" + id + "' id='photo_id' name='photo_id'/>");
-        } else {
-          oldPhoto.val(id);
-        }
+          /* flash message prompt */
+          var message = Diaspora.I18n.t("photo_uploader.looking_good");
+          if (app.flashMessages) { app.flashMessages.success(message); }
 
-        $("#profile_photo_upload").find(".avatar").removeClass("loading");
-        $("#profile_photo_upload").find(".avatar").attr("src", url);
+          id = responseJSON.data.photo.id;
```

Comment:
When the upload failed:

```
[Fine Uploader 5.11.0] Caught exception in 'onComplete' callback - responseJSON.data is undefined
```

Please make sure that you only do these changes when the upload was successful.

(optional)